glib/src/functions.rs

```diff
@@ -212,3 +210,17 @@ pub fn spawn_async_with_pipes<
         }
     }
 }
+
+/// Obtain the character set for the current locale.
+///
+/// This returns whether the locale's encoding is UTF-8, and the current
+/// charset if available.
+pub fn get_charset() -> (bool, Option<GString>) {
+    let (is_utf8, charset);
+    let mut out_charset = ptr::null();
+    unsafe {
+        is_utf8 = from_glib(glib_sys::g_get_charset(&mut out_charset));
+        charset = from_glib_none(out_charset);
+    };
+    (is_utf8, charset)
```

Comment:
Now if you move the tuple inside the `unsafe` block, you can avoid that spurious variable declaration at the top :)

Personally I'd just put the whole function into a single unsafe block. Everything it does is unsafe apart from actually returning the values, but those you have to get out of the unsafe block anyway so you can just return them from there.

Looks good otherwise, thanks :)